synapse/handlers/federation.py

```diff
@@ -294,6 +294,15 @@ def backfill(self, dest, room_id, limit, extremities=[]):
             extremities=extremities,
         )
 
+        seen_events = yield self.store.have_events_in_timeline(
+            set(e.event_id for e in events)
+        )
```

Comment:
Could you add a comment explaining why this is needed?
Also possibly elaborate on what the behaviour of the caller will be?